File: JPA_ONE-TO-ONE/src/main/java/com/one/controller/UserController.java
package com.one.controller;

import com.one.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * created by Gyunny 2022/02/19
 */
@RequiredArgsConstructor
@RestController
public class UserController {

    private final UserService userService;

    @GetMapping("/user")
    public String user() {
        userService.test();
        return "test";
    }
}

File: Spring_Session/src/main/kotlin/com/session/controller/UserController.kt
package com.session.controller

import com.session.dto.SignInDTO
import com.session.dto.SignUpDTO
import com.session.service.UserService
import org.springframework.format.annotation.DateTimeFormat
import org.springframework.web.bind.annotation.*
import java.time.LocalDate
import javax.servlet.http.HttpSession

/**
 * created by Gyunny 2021/10/18
 */
@RequestMapping("/api/v1")
@RestController
class UserController(
    private val userService: UserService
) {

    @GetMapping
    fun test(httpSession: HttpSession): String {
        httpSession.setAttribute("key", "value")
        return "test"
    }

    @GetMapping("/{date}")
    fun bind(@DateTimeFormat(pattern = "yyyy-MM-dd") @PathVariable("date") localDate: LocalDate): String {
        println("Test")
        return "test"
    }

    @GetMapping("/test")
    fun test1(httpSession: HttpSession): String {
        println(httpSession.getAttribute("key"))
        return "hi"
    }

    @PostMapping("/signup")
    fun sign(@RequestBody signUpDTO: SignUpDTO, httpSession: HttpSession) {
        httpSession.setAttribute("key", signUpDTO.email)
        println(httpSession.getAttribute("key"))
        userService.signup(signUpDTO)
    }

    @PostMapping("/signin")
    fun login(@RequestBody signInDTO: SignInDTO, httpSession: HttpSession) {
        println("=====")
        println(httpSession.getAttribute("key"))
        println("=====")
        userService.signIn(signInDTO)
    }

}